src/main/java/github/grace000/erversay/RequestRouter.java

```diff
@@ -4,17 +4,25 @@
 
 import java.util.HashMap;
 
+import static github.grace000.erversay.Constants.StatusCodes.NOT_FOUND_STATUS;
+
 public class RequestRouter {
     public String route(Request request, HashMap routes) {
         return handleRequest(request, routes);
     }
 
     private String handleRequest(Request request, HashMap routes) {
         String path = request.path;
-        if (routes.containsKey(path)) {
-            RouteHandler routeHandler = (RouteHandler) routes.get(path);
-            return routeHandler.handle(request);
+
+        if (!routes.containsKey(path)) {
+            return handleUnknownPath();
         }
-        return path;
+
+        RouteHandler routeHandler = (RouteHandler) routes.get(path);
```

Comment:
Looks good, although is there are reason you're casting to `RouteHandler` when the `HashMap` has a generic type of `<String, RouteHandler>`? Could you make a slight tweak to the parameters in the method's signature?

Reply:
I first tried this implementation, but I wasn't able to use the handle method without casting to RouteHandler:
`private String handleRequest(Request request, HashMap routeMap) {
        String path = request.path;

        if (!routeMap.containsKey(path)) {
            return handleUnknownPath();
        }

        RouteHandler routeHandler = routeMap.get(path);
        return routeHandler.handle(request);
    }`